avocado/core/task/statemachine.py

```diff
@@ -247,6 +247,7 @@ async def start(self):
             await asyncio.sleep(0.1)
             return
 
+        await self._spawner.create_task_output_dir(runtime_task)
```

Comment:
I don't want to do that, because IMO this will break the `Spawner` interface. If I put this into the `spawn_task()` then the `spawn_task()` can't be an abstract method. And I need to call `spawner.create_task_output_dir` before the `spawner.spawn_task()` but that is something I can't do it inside Spawner itself without braking the interface. If I do that, it will create a new requirement for development of new Spawners and I would like to avoid it.

Reply:
I see, the problem with the current approach is that we are adding spawner logic inside the statemachine, what I think is not also a good idea.

Reply:
I try to fix this problem by creating method `spawn_task_with_output` inside Spawner interface. I am not really sure if this is the right fix, but I wasn't able to find better solution.